scripts/download_ngtcp2.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# If any commands fail, fail the script immediately.
+set -ex
+
+# Clone the repository to the specified directory.
+git clone --depth 1 --branch v1.1.0 https://github.com/ngtcp2/ngtcp2 $1
+
+# Teach ngtcp2 about sockets
+pushd $1
+patch -p1 <<'EOF'
```

Comment:
Inlining the patch here is nasty. I'd rather this were in a patch file.

Reply:
Also, doing this via a patch mechanism severely limits updating this dependency in the future. What's the long-term plan here - add support to ngtcp2 upstream?

Reply:
Yeah, having the patch here is not a permanent solution. The long-term plan would be to work with ngtcp2 to make it more fuzzing friendly -- e.g. by allowing it to work with AF_UNIX sockets like this patch does.

Reply:
@bagder do you have an opinion here? In my experience this kind of mechanism is fairly brittle; I don't know if it's something we want to support moving forwards.